Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferParserAVFObjC.mm

```diff
@@ -241,11 +241,11 @@ String extractCodecName(AVAssetTrack* track)
     [m_delegate invalidate];
 }
 
-Expected<void, PlatformMediaError> SourceBufferParserAVFObjC::appendData(Segment&& segment, AppendFlags flags)
+Expected<void, PlatformMediaError> SourceBufferParserAVFObjC::appendData(Ref<const SharedBuffer>&& segment, AppendFlags flags)
 {
     INFO_LOG_IF_POSSIBLE(LOGIDENTIFIER);
-    auto sharedBuffer = segment.takeSharedBuffer();
-    auto nsData = sharedBuffer->makeContiguous()->createNSData();
+    Ref sharedBuffer = WTFMove(segment);
```

Comment:
Why do we need to move segment?
Can we do:
`RetainPtr nsData = segment->createNSData()`?
Maybe we could pass a `const&` depending on the other implementations

Reply:
To control the lifecycle and ensuring the segment is deleted on the work queue.